### ISSUE TITLE
Reinstall option is missing in Windows installer

### ISSUE THREAD

**Issue by mikunimaru** [google/mozc#734]
**Description**
When making subtle changes to the code and regenerating the installer for Windows, it's not possible to reinstall Mozc because the installer doesn't recognize the update.

**Steps to reproduce**
1. Install Mozc from the installer for Windows
2. Update the version of Qt and build the installer again
3. Launch the regenerated installer.

**Expected behavior**
Provide an option for reinstall or overwrite installation of Mozc.

**Actual behavior**
![mozc](https://user-images.githubusercontent.com/43168745/234218958-fa32271e-a0d7-49b8-a265-c19dccc5fc29.jpg)

**Version or commit-id**
Mozc-2.28.5080.100+24.11.oss

**Environment**
 - OS: Windows

**Comment by yukawa** [comment]
Right.  The current behavior is what we have inherited from Google Japanese Input's installer, which was written more than 13.5 years ago. Since then the logic hasn't been changed much.

The relevant code is here.
https://github.com/google/mozc/blob/4c85ed6d0a6eb97e21d57adb0aecc87da0e6453c/src/win32/installer/installer_oss_64bit.wxs#L317-L325

I think the easiest (and the safest) way to achieve your request is to change the above lines as follows.
```XML
    <Upgrade Id='$(var.UpgradeCode)'>
      <UpgradeVersion
          Minimum='0.0.0.0' IncludeMinimum='yes'
          Maximum='$(var.MozcVersion)' IncludeMaximum='yes'
          OnlyDetect='no' Property='UPGRADING'/>
      <UpgradeVersion
          Minimum='$(var.MozcVersion)' IncludeMinimum='no'
          OnlyDetect='yes' Property='NEWERVERSIONDETECTED' />
    </Upgrade>
```

Perhaps we can now switch from [`<Upgrade>`](https://wixtoolset.org/docs/v3/xsd/wix/upgrade/) to [`<MajorUpgrade>`](https://wixtoolset.org/docs/v3/xsd/wix/majorupgrade/), which was added in WiX 3.5, but we really need to be careful about when [`RemoveExistingProducts`](https://learn.microsoft.com/ja-jp/windows/win32/msi/removeexistingproducts-action) action is scheduled when doing such a migration. So probably just flipping `IncludeMaximum` and `IncludeMinimum` would be safer for now.

**Comment by yukawa** [comment]
Submitted the above change internally. Should be fixed in the next code push.
